homeassistant/components/template/weather.py

```diff
@@ -76,6 +81,11 @@
         vol.Optional(CONF_VISIBILITY_TEMPLATE): cv.template,
         vol.Optional(CONF_FORECAST_TEMPLATE): cv.template,
         vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_TEMPERATURE_UNIT): cv.string,
+        vol.Optional(CONF_PRESSURE_UNIT): cv.string,
+        vol.Optional(CONF_WIND_SPEED_UNIT): cv.string,
+        vol.Optional(CONF_VISIBILITY_UNIT): cv.string,
+        vol.Optional(CONF_PRECIPITATION_UNIT): cv.string,
```

Comment:
VALID_UNITS_TEMPERATURE
```suggestion
        vol.Optional(CONF_TEMPERATURE_UNIT): vol.In(VALID_UNITS_TEMPERATURE),
        vol.Optional(CONF_PRESSURE_UNIT): cv.vol.In(VALID_UNITS_PRESSURE),
        vol.Optional(CONF_WIND_SPEED_UNIT): cv.vol.In(VALID_UNITS_WIND_SPEED),
        vol.Optional(CONF_VISIBILITY_UNIT): cv.vol.In(VALID_UNITS_VISIBILITY),
        vol.Optional(CONF_PRECIPITATION_UNIT): cv.vol.In(VALID_UNITS_PRECIPITATION),
```